### PR TITLE
Nessus plugin - add search to relevant commands

### PR DESCRIPTION
**Pull request by sempervictus**
Utilize the SearchTerm param in Rex' Tables to filter scrolling
outputs.
Clean up argument parsing (implement one of the MSF standards).
Update help commands to reflect changes

Testing: basic functional tests only so far

### CODE REVIEW

**Comment by void-in** [comment]
@sempervictus That is actually a great contribution. When I rewrote that plugin almost a year back, the Nessus API changed from XML to JSON and the focus was to make the plugin workable with the latest version. It is great you focused on an area which don't get the love of devs!

**Comment by sempervictus** [comment]
Thanks, i try to please, figured you'd see this rather quickly (lurker).
You beat me to publishing the plugin by a short time, i actually wrote a whole API gem just for that ;).
#1621 was basically the same thing as this PR a few years ago.
For historical reference, getting the 'SearchTerm' bit adopted upstream took some work, and nobody really uses it - it originally came as a "workable alternative" to efforts by Kernelsmith when we were still entertaining dreams of bash-like pipes and such in console.
I use Nessus quite a bit, figured since i was doing some work on my PSH code (or rather bug-hunting framework module instantiation ghosts), i'd push another useful bit upstream.

File: plugins/nessus.rb [hunk 1]
```diff
@@ -370,15 +423,23 @@ def cmd_nessus_save(*args)
...
-          print_status("Example:> nessus_server_feed")
-          print_status()
-          print_status("Returns information about the feed type and server version.")
-          return
+        search_term = nil
+        while (arg = args.shift)
+          case arg
+          when '-h', '--help'
+            print_status("nessus_server_feed")
+            print_status("Example:> nessus_server_feed -S searchterm")
```
Comment (void-in) [inline plugins/nessus.rb]: This was wrong before and it is wrong here as well. The command is nessus_server_properties but the example is nessus_server_feed. We need to change either the command or the example description.

Comment (sempervictus) [inline plugins/nessus.rb]: Addressed, thank you.

File: plugins/nessus.rb [hunk 2]
```diff
@@ -410,13 +479,19 @@ def cmd_nessus_server_status(*args)
...
+          when '-h', '--help'
+            print_status("nessus_admin")
+            print_status("Example:> nessus_admin")
+            print_status()
+            print_status("Checks to see if the current user is an admin")
+            print_status("Use nessus_user_list to list all users")
+            return
+          end
+        end
         if args[0] == "-h"
```
Comment (jlee-r7) [inline plugins/nessus.rb]: Body is empty, delete this line.

Comment (sempervictus) [inline plugins/nessus.rb]: I think the original author placed that there for formatting. Void-in: can
I nix the formatting line(s)?

Comment (void-in) [inline plugins/nessus.rb]: @sempervictus Yes. It should be removed.

**Comment by jlee-r7** [comment]
@sempervictus Ah, I see where the confusion was now. That's not what I meant. The `if` block on that line has no body so it is a pointless comparison.

**Comment by jlee-r7** [comment]
Sorry I wasn't more clear

**Comment by void-in** [comment]
@jlee-r7 Yeah. I can see a confusion now but I think the latest commit shouldn't be reverted. The call to print_status() was included just for an empty line which isn't needed. Only thing left is to remove https://github.com/sempervictus/metasploit-framework/blob/ae38403bc29ec20c32f2cefeea8d3bd6f2990c4e/plugins/nessus.rb#L490-L492

**Comment by wvu-r7** [comment]
@void-in: Are you taking this?

**Comment by void-in** [comment]
@wvu-r7 Please land this. I am away from my dev machine and will take another 10+ hours to get there. Sorry I didn't know it was assigned to me.

**Comment by sempervictus** [comment]
Thx guys. I've been using this a ton and its turned out extremely helpful
when trolling thru large reports in console. Actually easier than the GUI
or my pitiful api gem.

**Comment by void-in** [comment]
Thanks @wvu-r7 for landing. Great job @sempervictus

**Pull request merged**